controllers/openstackcontrolplane_controller.go

```diff
@@ -480,3 +481,21 @@ func (r *OpenStackControlPlaneReconciler) setProvisioningStatus(instance *ospdir
 
 	return nil
 }
+
+func (r *OpenStackControlPlaneReconciler) createUniqNetworkList(instance *ospdirectorv1beta1.OpenStackControlPlane) []string {
+
+	// create uniq list networls of all VirtualMachineRoles
+	networkList := make(map[string]int)
+	for _, vmRole := range instance.Spec.VirtualMachineRoles {
+		for _, network := range vmRole.Networks {
+			networkList[network] = 1
+		}
+	}
```

Comment:
Ah, you use a map here to avoid duplicates and then turn it into a list below.  Clever!